### DISCUSSION TITLE
Symmetrization of Hamiltonian for 2D material

### DISCUSSION THREAD

**Discussion by Bhuvaneswari-R432** [wannier-berri/wannier-berri#236] [Q&A] [answered]
Kindly let me know if symmetrization of Hamiltonian for a 2D material possible? When I am trying to symmetrize the Hamiltonian for a 2D system, I encounter the following error:

test_i = self.iRvec.index([0, 0, 1])
ValueError: [0, 0, 1] is not in list.

However, it is running fine for a bulk 3D system. Please let me know how to rectify the error.

Regards,
Bhuvaneswari R

**Comment by stepan-tsirkin** [comment]
@Bhuvaneswari-R432 , could you provide some more details on your calculation, the python script that you are executing and a complete stack trace of the error.

**Reply by Bhuvaneswari-R432** [reply]
Sir, I am working on a 2D phosphorene bilayer (without soc) using quantum espresso (with s and p as projections). I provided positions (of atoms), atom_name, proj, soc and DFT_code in the symmetrize() function. Symmetrization if finished and while testing AA with diag = true, [0,0,0], [1,0,0] and [0,1,0] is fine and I am getting the following error.

File "symmetrization.py", line 20, in
DFT_code='espresso',
File "/usr/local/lib/python3.6/dist-packages/wannierberri/system/system.py", line 146, in symmetrize
XX_R, self.iRvec = symmetrize_wann.symmetrize()
File "/usr/local/lib/python3.6/dist-packages/wannierberri/system/sym_wann.py", line 530, in symmetrize
test_i = self.iRvec.index([0, 0, 1])
ValueError: [0, 0, 1] is not in list.

The python script is given here: https://github.com/wannier-berri/wannier-berri/files/10862050/symmetrization.zip
Kindly help me with it. Thank you.

Regards,
Bhuvaneswari R

**Comment by Bhuvaneswari-R432** [comment]
Sir, I am working on a 2D phosphorene bilayer (without soc) using quantum espresso (with s and p as projections). I provided positions (of atoms), atom_name, proj, soc and DFT_code in the symmetrize() function. Symmetrization if finished and while testing AA with diag = true, [0,0,0], [1,0,0] and [0,1,0] is fine and I am getting the following error.

File "symmetrization.py", line 20, in <module>
    DFT_code='espresso',
  File "/usr/local/lib/python3.6/dist-packages/wannierberri/system/system.py", line 146, in symmetrize
    XX_R, self.iRvec = symmetrize_wann.symmetrize()
  File "/usr/local/lib/python3.6/dist-packages/wannierberri/system/sym_wann.py", line 530, in symmetrize
    test_i = self.iRvec.index([0, 0, 1])
ValueError: [0, 0, 1] is not in list.

Kindly help me with it. Thank you.

Regards,
Bhuvaneswari R

**Reply by Bhuvaneswari-R432** [reply]
[symmetrization.zip](https://github.com/wannier-berri/wannier-berri/files/10862050/symmetrization.zip)
Dear sir,
This is the python script I used to symmetrize. 

Regards,
Bhuvaneswari R

**Reply by stepan-tsirkin** [reply]
you error is caused by line 530 of `sym_wann.py`, but the current version of the [file](https://github.com/wannier-berri/wannier-berri/blob/master/wannierberri/system/sym_wann.py) has only 521 lines. Please, upgrade to the latest version of WannierBerri and tell me what comes out.

**Reply by Bhuvaneswari-R432** [reply]
Dear sir,

After upgrading wannierberri to its latest version, symmetrization is perfectly done for 2D material. Thank you.

Regards,
Bhuvaneswari R

**Comment by stepan-tsirkin** [comment, marked as answer]
The  error is caused by line 530 of `sym_wann.py`, but the current version of the [file](https://github.com/wannier-berri/wannier-berri/blob/master/wannierberri/system/sym_wann.py) has only 521 lines. Please, upgrade to the latest version of WannierBerri
